Event/HreftypeaheadSearchEvent.php

```diff
@@ -0,0 +1,66 @@
+<?php
+/**
+ * Pimcore
+ *
+ * This source file is available under two different licenses:
+ * - GNU General Public License version 3 (GPLv3)
+ * - Pimcore Enterprise License (PEL)
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ * @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ * @license    http://www.pimcore.org/license     GPLv3 and PEL
+ */
+
+namespace PimcoreHrefTypeaheadBundle\Event;
+
+use Pimcore\Model\DataObject;
+use Symfony\Component\EventDispatcher\Event;
+
+class HreftypeaheadSearchEvent extends Event
+{
+    /** @var DataObject\Conrete */
+    protected $sourceObject;
+
+    /** @var array */
+    protected $conditions;
+
+    /**
+     * HreftypeaheadSearchEvent constructor.
+     *
+     * @param array $conditions
+     * @param array $arguments
+     */
```

Comment:
Why is `&$conditions` passed by reference? And if the type of the array is known (strings?) then setting the docblock to `string[] $conditions` will make IDE's happy.

Reply:
pass by ref so you can add things to the original array